Riot/Managers/Analytics/AnalyticsSettings.swift

```diff
@@ -0,0 +1,61 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+struct AnalyticsSettings {
+    static let eventType = "im.vector.analytics"
+    
+    private enum Constants {
+        static let idKey = "id"
+        static let webOptInKey = "pseudonymousAnalyticsOptIn"
+    }
+    
+    /// A randomly generated analytics token for this user.
+    /// This is suggested to be a 128-bit hex encoded string.
+    var id: String?
+    
+    /// Unused on iOS but necessary to load the value in case opt in was declined on web,
+    /// but accepted on iOS. Otherwise generating an ID would wipe out the existing value.
+    private var webOptIn: Bool?
```

Comment:
I've updated the doc comment on this, but yes it needs to be nullable:

> `true` if opted in on web, `false` if opted out on web and `nil` if the web prompt is not yet seen.

I left it private to avoid any confusion as we do not use this value on iOS.